### DISCUSSION TITLE
What's the worst that could happen to my photos

### DISCUSSION THREAD

**Discussion by Coniface** [immich-app/immich#9974] [General]
I was thinking of installing Immich but this very disclaimer highly dissuades me to:

> ⚠️ Do not use the app as the only way to store your photos and videos.

I tried to find more explanations as to what are the risks and how I could mitigate them with no luck.

If I installed it and used it as "the only way to store all my photos", then I backup these photos using a 3-2-1 plan, what is the worst that could happen that would explain the reason for this disclaimer?

Can the program somehow overwrite the pictures? Can the container delete the files? Is it only about the database state of the pictures? Would the files be safe by themselves?

**Comment by eerison** [comment]
atm I am just using immich as readonly on docker volumes, after I have tested the app completely, then I gonna try to use this to manager my photos like remove if necessary.

But using the app just to visualise and share photos is working really good for me :)

**Comment by j-seixas** [comment]
That really is just a disclaimer for using a 3-2-1 backup plan and not relying only in immich. If you have it as **the only way to store all photos**, then it isn't _really_ a **3-2-1**.. Also the points above explain a bit more on the why:

> - The project is under very active development.
> -  Expect bugs and breaking changes.

So since there may have breaking changes and bugs, there is a strong possibility of encountering one. I would say that currently its pretty stable and unless you are changing or messing around configs nothing should happen to your photos/videos. Just be careful with updating versions and check the changelog if there are any breaking changes or config changes that you may need to do.

Don't forget that if your hard drive / machine goes kapoot, you may lose all media. That's why a 3-2-1 is recommended as a general rule and not only for "immich". Same thing if somehow there is a fire in google servers and all your data gets lost

Also to mitigate possible issues and media loss you can:
- 3-2-1 backup of course, not relying on only immich, save your photos also in a external hard drive for example, somewhere in the cloud or in some other physical location for redundancy, etc
- backup periodically your immich db (there is a guide in the documentation) -> you can also backup your entire media folders (using rsync for example)

**Comment by vincios** [comment]
At this stage of development, i think that disclaimer mainly concerns the upload library and the mobile [automatic backup](https://immich.app/docs/features/automatic-backup) feature.

Imagine this scenario: you rely on the backup feature of Immich's mobile app to backup your phone's images, then you set up a 3-2-1 strategy (on the server) for backing up the Immich upload library.

The background backup feature of the moblie app is still buggy (for example, the Android OS often kills the background job if not properly configured).

Now immagine that a new update of the mobile app has broken the automatic backup: it shows you that "the backup is OK" even if, for some reason, not all your asset have been uploaded correctly. You might think "okay, I can safely delete them from my phone". So, you irrmediably lose all photos that were not uploaded correctly!

Another example has just happened with the latest release: a bug of the new "external library delete" feature [causes](https://github.com/immich-app/immich/issues/9460#issuecomment-2110645720) files on the storage to be deleted too. But, yeah, in this case a 3-2-1 backup should be enough to recover deleted assets.
